Many Unix MUAs require that a mail message header begin with a line that starts
with the five-character sequence “From ”.  For that reason, most MTAs strip
these header lines out of messages before SMTP delivery.
Without the following patch, DMA does not do this and that causes problems
with messages generated via UNIX utilities such as the mail command
(which happens to be used by cron on OmniOS).

diff -wpruN '--exclude=*.orig' a~/mail.c a/mail.c
--- a~/mail.c	1970-01-01 00:00:00
+++ a/mail.c	1970-01-01 00:00:00
@@ -399,7 +399,9 @@ readmail(struct dma_queue *queue, int no
 			if (!(line[0] == ' ' || line[0] == '\t'))
 				nocopy = 0;
 
-			if (strprefixcmp(line, "Date:") == 0)
+			if (strprefixcmp(line, "From ") == 0)
+				continue;
+			else if (strprefixcmp(line, "Date:") == 0)
 				had_date = 1;
 			else if (strprefixcmp(line, "Message-Id:") == 0)
 				had_messagid = 1;
